project-koku/koku
# COST-2831 - clear celery queue

## Jira Ticket

[COST-2831](https://issues.redhat.com/browse/COST-2831)

## Description

This change will add a masu endpoint to purge tasks in the celery queues

## Testing

1. Checkout Branch
2. Restart Koku
3. run `make docker-reinitdb-with-sources-lite`
4. Hit these endpoints after
    i. check current queue lengths
    - `http://127.0.0.1:5042/api/cost-management/v1/celery_queue_lengths/`
     - You should see a dictionary with format `{queue_name: queue_length}`

    ii.  to clear tasks in default celery queue
    - `http://127.0.0.1:5042/api/cost-management/v1/clear_celery_queues/`
    - you should see `{'purged_tasks': ##}`

    iii. to clear all task in all queues:     
    - `http://127.0.0.1:5042/api/cost-management/v1/clear_celery_queues/?clear_all=True`
    -  you should see `{'purged_tasks': ##}`


## Notes

...
